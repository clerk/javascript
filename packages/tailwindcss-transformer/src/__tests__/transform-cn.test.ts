import { transform } from '../index';

test('cn', () => {
  const result = transform(
    `const Button: React.FC<ButtonProps> = ({ children, className }) => {
    const buttonClasses = cn(
      (something === 'flex' || 'flex') && 'flex',
      ['flex', 'flex'],
      something === 'flex' || something === 'flex' ? (something ? 'flex' : 'flex') : 'flex',
      className,
    );
    return <button className={buttonClasses}>{children}</button>;
  };`,
    {
      styleCache: new Map(),
    },
  );
  expect(result).toMatchSnapshot();
});
