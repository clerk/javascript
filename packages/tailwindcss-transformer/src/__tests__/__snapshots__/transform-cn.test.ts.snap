// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cn 1`] = `
"const Button: React.FC<ButtonProps> = ({ children, className }) => {
    const buttonClasses = cn(
      (something === 'flex' || 'flex') && "cl-222f930b",
      ["cl-222f930b", "cl-222f930b"],
      something === 'flex' || something === 'flex' ? (something ? "cl-222f930b" : "cl-222f930b") : "cl-222f930b",
      className,
    );
    return <button className={buttonClasses}>{children}</button>;
  };"
`;
